Essentials/src/com/earth2me/essentials/signs/SignEnchant.java

```diff
@@ -70,18 +68,17 @@ protected boolean onSignInteract(ISign sign, User player, String username, IEsse
         final Trade charge = getTrade(sign, 3, ess);
         charge.isAffordableFor(player);
         final String[] enchantLevel = sign.getLine(2).split(":");
-        if (enchantLevel.length != 2) {
-            throw new SignException(tl("invalidSignLine", 3));
-        }
         final Enchantment enchantment = Enchantments.getByName(enchantLevel[0]);
         if (enchantment == null) {
             throw new SignException(tl("enchantmentNotFound"));
         }
-        int level;
-        try {
-            level = Integer.parseInt(enchantLevel[1]);
-        } catch (NumberFormatException ex) {
-            level = enchantment.getMaxLevel();
+        int level = 1;
+        if (enchantLevel.length == 2) {
+            try {
+                level = Integer.parseInt(enchantLevel[1]);
+            } catch (NumberFormatException ex) {
+                level = enchantment.getMaxLevel();
```

Comment:
I don't like this default. I know you didn't do it, but this just feels clowny.

Reply:
Would you rather it be 1?

Reply:
This currently retains the previous behaviour, where if the sign included a level but it was invalid, the sign would enchant at the max level. However, this case shouldn't ever get reached, as the sign is validated in `onSignCreate` above. If we've managed to get to this point, someone has deliberately bypassed validation, and I'd rather throw an error here than make assumptions about what the sign should do.